src/NServiceBus.Core/Transports/Msmq/CorrelationIdMutatorForBackwardsCompatibilityWithV3.cs

```diff
@@ -1,28 +0,0 @@
-﻿namespace NServiceBus.Transports.Msmq
-{
-    using System;
-    using MessageMutator;
-    using Unicast.Messages;
-
-    [ObsoleteEx(RemoveInVersion ="6.0")]
-    class CorrelationIdMutatorForBackwardsCompatibilityWithV3 : IMutateOutgoingTransportMessages
```

Comment:
Have not reviewed this PR yet, so not sure
But the things to watch out is breaking wire compat, that is all IMO.